pkgs/applications/networking/wgnord/default.nix

```diff
@@ -0,0 +1,67 @@
+{ bash
+, coreutils
+, curl
+, fetchFromGitHub
+, gnugrep
+, gnused
+, iproute2
+, jq
+, lib
+, makeWrapper
+, resholve
+, wireguard-tools
+}:
+
+resholve.mkDerivation rec {
+  pname = "wgnord";
+  version = "0.1.10";
+
+  src = fetchFromGitHub {
+    owner = "phirecc";
+    repo = pname;
+    rev = version;
+    hash = "sha256-T7dAEgi4tGvrzBABGLzKHhpCx0bxSCtTVI5iJJqJGlE=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace wgnord \
+      --replace '$conf_dir/countries.txt' "$out/share/countries.txt" \
+      --replace '$conf_dir/countries_iso31662.txt' "$out/share/countries_iso31662.txt"
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    install -Dm755 wgnord -t $out/bin/
+    install -Dm 644 countries.txt -t $out/share/
+    install -Dm 644 countries_iso31662.txt -t $out/share/
+  '';
+
+  solutions.default = {
+    scripts = [ "bin/wgnord" ];
+    interpreter = "${bash}/bin/bash";
+    inputs = [
+      coreutils
+      curl
+      gnugrep
+      gnused
+      iproute2
+      jq
+      wireguard-tools
+    ];
+    execer = [
+      "cannot:${iproute2}/bin/ip"
+      "cannot:${wireguard-tools}/bin/wg-quick"
+    ];
```

Comment:
this works, and despite coming from [a gist](https://gist.github.com/PedroHLC/e6cefa4dce6474d27f3a91f4ff929a67) and feeling like a bodge, seems canonical based on [upstream comments](https://github.com/abathur/resholve/issues/77#issuecomment-1076494874)?

@abathur: can you assist me in the "ideal outcome", and show me how to "improve resholve and/or binlore", as I am keen "to fix it for everyone"?

Reply:
It is, (but I did the same in my diff: https://github.com/abathur/nixpkgs/commit/d6030d4be4a6c8fb084f617d9991b5211405596e#diff-30325ce60c431ba4328f918690dc08e9999399684a0747ba4af0e9648c238bc4R52-R55).

I don't have a great guide on these yet (though I did start one at some point; trying to defer this work until someone's asking so that it can be shaped by feedback).

We can discuss this elsewhere (Matrix? resholve's repo?), but the broad strokes are:
1. If binlore isn't handling the language/format these executables are in, we can add/improve its YARA rules
2. binlore uses very loose heuristics to look for any exec behavior, so if it *is* handling the language/format, we need to look at the source to disambiguate.
    - If it isn't actually user-controlled from the CLI, we can set an override in binlore to ~whitelist it.
    - Since these overrides are heavy-handed/fragile (the executable could gain exec in some future version), it would be _extremely_ ideal to lay down some practices for getting the assumptions from hand audits under test. (No expectation here. Just a task that needs doing. I have some WIP somewhere, but I haven't had time to revisit it for a while.)
3. If it *does* have user-controlled exec behavior, resholve needs a parser that can identify invocations that use it. This is some mix of collecting research and writing a parser.

Reply:
yeah, your call on if you want to spawn a ticket based on this in resholve or hit me up on matrix ([info](https://github.com/NixOS/nixpkgs/blob/d87fcbe3cd948f8f1616723d2ed2e05f13ca13a5/maintainers/maintainer-list.nix#L14561) in the maintainers file); I will resolve this to ensure that there is no confusion about the status of this pr